File: lab 7/scala/src/main/scala/NaivePhilosopher.scala
class NaivePhilosopher(val number: Int, val f1: Fork, val f2: Fork)
  extends Philosopher(number, f1, f2)
{

  @throws[InterruptedException]
  override protected def eat(): Unit = {
    super.consume(f1, f2)
  }
}

File: lab 7/java/src/NaivePhilosopher.java
import java.util.Random;

public class NaivePhilosopher extends Philosopher
{
    public NaivePhilosopher(int number, Fork f1, Fork f2)
    {
        this.number = number;
        this.f1 = f1;
        this.f2 = f2;
    }

    protected void eat() throws InterruptedException
    {
        super.consume(f1, f2);
    }
}
